### PR TITLE
Add lsp-find-definition-mouse.

### LINKED ISSUE #564: [mouse]  Right click support  for lsp-mode
1. Right click support
2. Ctrl + click - to follow references
3. ?

### PR DESCRIPTION
**Pull request by leungbk**
Related to #564.

Based on http://ergoemacs.org/emacs/emacs_mouse_click_highlight_word.html.

### CODE REVIEW

File: lsp-mode.el [hunk 1]
```diff
@@ -3984,6 +3984,15 @@ REFERENCES? t when METHOD returns references."
...
   (lsp-find-locations "textDocument/definition" nil :display-action display-action))
 
+(defun lsp-find-definition-mouse (click)
+  "Click to start `lsp-find-definition' at clicked point."
+  (interactive "e")
+  (let ((p1 (posn-point (event-start click))))
+    (goto-char p1)
+    (lsp-find-definition)))
+
+(define-key lsp-mode-map (kbd "C-<down-mouse-1>") #'lsp-find-definition-mouse)
```
Comment (yyoncho) [inline lsp-mode.el]: Can you group that in the initialization option, like that: 

```
(defvar tide-mode-map
  (let ((map (make-sparse-keymap)))
    (define-key map (kbd "M-.") #'tide-jump-to-definition)
    (define-key map (kbd "M-,") #'tide-jump-back)
    map))
```
AFAIK this is the standart pattern to initialize a modemap.

**Comment by leungbk** [comment]
OK, I've made your suggested changes.

**Comment by yyoncho** [comment]
Thank you and congrats for your first `lsp-mode` PR!

**Comment by myrgy** [comment]
Great feature! Thanks a lot!

**Pull request merged**